user-auth-api.py

```
@@ -0,0 +1,108 @@
+@app.route('/admin/users', methods=['GET'])
+def get_users():
+    # Broken access control - no authorization check
+    if 'user_id' in session:
+        conn = mysql.connector.connect(**DB_CONFIG)
+        cursor = conn.cursor()
+        cursor.execute("SELECT * FROM users")
+        users = cursor.fetchall()
+        
+        # Unsafe deserialization
+        if request.args.get('filter'):
+            filter_obj = pickle.loads(request.args.get('filter').encode())
```

Ruff (0.12.2)

Local variable `filter_obj` is assigned to but never used

Remove assignment to unused variable `filter_obj`

(F841)